src/Servers/HttpSys/src/RequestProcessing/RequestContext.cs

```diff
@@ -63,7 +63,7 @@ public CancellationToken DisconnectToken
                         }
                         else
                         {
-                            _disconnectToken = CancellationToken.None;
+                            _disconnectToken = new CancellationToken(canceled: true);
```

Comment:
When exactly `DisconnectListener` return `CancellationToken.None`? Should we just update `DisconnectListener.GetTokenForConnection(connectionId)` to return `new CancellationToken(canceled: true)` in whatever that case is? It looks like it already does that for most errors but not all.

Reply:
The main example is here:
https://github.com/dotnet/aspnetcore/blob/d2421adbc9b7472df187ffdc71374cd4b253508e/src/Servers/HttpSys/src/NativeInterop/DisconnectListener.cs#L33-L37

I wasn't able to force this condition in testing, and it's not directly related to the original issue. The primary fix for the IOException issue is in Abort. I'll revert this part until we get a repro.

```suggestion
                            _disconnectToken = CancellationToken.None;
```